src/software/ai/hl/stp/play/play.cpp

```diff
@@ -65,54 +77,215 @@ PriorityTacticVector Play::getTactics(const World &world)
     return PriorityTacticVector();
 }
 
-std::vector<std::unique_ptr<Intent>> Play::get(
-    RobotToTacticAssignmentFunction robot_to_tactic_assignment_algorithm,
-    MotionConstraintBuildFunction motion_constraint_builder, const World &new_world,
+std::unique_ptr<TbotsProto::PrimitiveSet> Play::get(
+    const GlobalPathPlannerFactory &path_planner_factory, const World &world,
```

Comment:
nit: this is a 200 line function, can we maybe have a private helper for the tactic assignment part and call it here?

Reply:
I'll look into it

Reply:
ok refactored